tools/cloudflow-descriptor-generator/src/main/scala/akka/cloudflow/StreamletScanner.scala

```diff
@@ -86,16 +96,25 @@ object StreamletScanner {
       case _ => getInstanceFromDefaultConstructor(clazz)
     }
 
-  private def getInstanceFromDefaultConstructor(streamletClass: Class[_]): Try[Any] =
-    if (!hasDefaultConstructor(streamletClass)) {
-      Failure(ConstructorMissing(streamletClass))
-    } else {
-      Try(streamletClass.getDeclaredConstructor().newInstance()).recoverWith {
-        case error => Failure(ConstructorFailure(streamletClass, error))
-      }
+  private def getInstanceFromDefaultConstructor(streamletClass: Class[_]): Try[Any] = {
+    getNoArgConstructor(streamletClass) match {
+      case Some(constructor) =>
+        try {
+          constructor.newInstance()
+        } catch {
+          case ex: Throwable => ex.getCause.printStackTrace()
```

Comment:
This does not seem right?

Reply:
Would have preferred to leave the original code as is.. I think in this case before it would result in a Failure?

Reply:
Another thing, that has been on some list.. we should prevent people from defining a Streamlet as `object`. It's not a problem on the cluster, but in local running it can be, if we fork a JVM per set of streamlets in the same library. (we can change that as well, so that we fork a jvm for every object, if needed, but just wanted to mention it FYI)

Reply:
is a leftover, fixing it